org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/projectWizard/TemplateProjectWizardFragment.xtend

```diff
@@ -0,0 +1,519 @@
+		import org.eclipse.xtext.ui.wizard.template.ProjectTemplate
+
+		import static org.eclipse.core.runtime.IStatus.*
+
+		/**
+		 * Provide all available project templates to be available in the NewProjectWizard. Each template is described through one class annotated with @Template.
+		 * With @Template the label, icon and description of the template, displayed to the user in the wizard, are defined.
+		 * 
+		 * In addition it is possible to define fields that are displayed as text field, checkbox or combobox. There exist three methods #text, #check and #combo
+		 * to create such fields. These fields are presented to the user on a second page and can be used to configure the generated content by the end user.
+		 * 
+		 * The variables can be enabled/disabled by overriding #updateVariables. Each variable has a #isEnabled and a #setEnabled. It is also allowed to change
+		 * the variable values in this method (e.g. to change a string in response to a change of a checkbox).
+		 * 
+		 * The variables can be validated by overriding the method #validate. Errors generated here are displayed in the wizard. The validation takes place after
+		 * #updateVariables has been executed.
+		 * 
+		 * In the end code is to be generated. The method #generateFiles is executed. The given IrojectFileGenerator can be used to generate files inside the
+		 * newly generated project. Values from variables might be used here to configure the generated files.
+		 * 
+		 * <pre>
+		 * 	def helloFile() {
+		 * 		new ProjectFile(�openVar�quotes�closeVar�src/�openVar�path�closeVar�/Model.pas�quotes�, �quotes�
+		 * 			Hello �openVar�name�closeVar�!
+		 * 		�openVar�quotes�closeVar�);
+		 * 	}
+		 * </pre>
+		 */
+		class �initialContentsClass.simpleName� implements IProjectTemplateProvider {
+			override getProjectTemplates() {
+				#[new HelloWorldProject]
+			}
+		}
+		
+		@ProjectTemplate(label="Hello World", icon="project_template.png", description="<p><b>Hello World</b></p>
+		<p>This is a parameterized hello world for �grammar.name�. You can set a parameter to modify the content in the generated file
+		and a parameter to set the package the file is created in.</p>")
+		final class HelloWorldProject {
+			var advanced = check("Advanced", false, "Check to enabled advanced configuration")
+			var name = combo("Name", #["Xtext", "World", "Foo", "Bar"], "The name to say 'Hello' to")
+			var path = text("Package", "�language.fileExtensions.get(0)�", "The package path to place the files in")
+		
+			override protected updateVariables() {
+				name.enabled = advanced.value
+				path.enabled = advanced.value
+				if (!advanced.value) {
+					name.value = "Xtext"
+					path.value = "�language.fileExtensions.get(0)�"
+				}
+			}
+		
+			override protected validate() {
+				if (path.value.matches('[a-z][a-z0-9_]*(/[a-z][a-z0-9_]*)*'))
+					null
+				else
+					new Status(ERROR, "Wizard", "'" + path + "' is not a valid package name")
```

Comment:
This suggests that '/' are valid package segments - path should replace all / by . in the error message.